x-pack/legacy/plugins/ml/public/timeseriesexplorer/components/timeseries_chart/timeseries_chart.js

```diff
@@ -1312,120 +1321,157 @@ const TimeseriesChartIntl = injectI18n(class TimeseriesChart extends React.Compo
+        tooltipData.push({
+          name: intl.formatMessage({
+            id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScore.predictionLabel',
+            defaultMessage: 'prediction'
+          }),
+          value: formatValue(marker.value, marker.function, fieldFormat),
+          seriesKey,
+          yAccessor: 'prediction'
         });
       } else {
-        contents += intl.formatMessage({
-          id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScore.valueLabel',
-          defaultMessage: 'value: {value}'
-        }, {
-          value: formatValue(marker.value, marker.function, fieldFormat)
+        tooltipData.push({
+          name: intl.formatMessage({
+            id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScore.valueLabel',
+            defaultMessage: 'value'
+          }),
+          value: formatValue(marker.value, marker.function, fieldFormat),
+          seriesKey,
+          yAccessor: 'value'
         });
       }
 
       if (modelPlotEnabled === true) {
-        contents += intl.formatMessage({
-          id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScoreAndModelPlotEnabled.upperBoundsLabel',
-          defaultMessage: '{br}upper bounds: {upperBoundsValue}'
-        }, {
-          br: '<br />',
-          upperBoundsValue: formatValue(marker.upper, marker.function, fieldFormat)
+        tooltipData.push({
+          name: intl.formatMessage({
+            id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScoreAndModelPlotEnabled.upperBoundsLabel',
+            defaultMessage: 'upper bounds'
+          }),
+          value: formatValue(marker.upper, marker.function, fieldFormat),
+          seriesKey,
+          yAccessor: 'upper_bounds'
         });
-        contents += intl.formatMessage({
-          id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScoreAndModelPlotEnabled.lowerBoundsLabel',
-          defaultMessage: '{br}lower bounds: {lowerBoundsValue}'
-        }, {
-          br: '<br />',
-          lowerBoundsValue: formatValue(marker.lower, marker.function, fieldFormat)
+        tooltipData.push({
+          name: intl.formatMessage({
+            id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.withoutAnomalyScoreAndModelPlotEnabled.lowerBoundsLabel',
+            defaultMessage: 'lower bounds'
+          }),
+          value: formatValue(marker.lower, marker.function, fieldFormat),
+          seriesKey,
+          yAccessor: 'lower_bounds'
         });
       }
     }
 
     if (_.has(marker, 'scheduledEvents')) {
-      contents += '<br/><hr/>' + intl.formatMessage({
-        id: 'xpack.ml.timeSeriesExplorer.timeSeriesChart.scheduledEventsLabel',
-        defaultMessage: 'Scheduled events:{br}{scheduledEventsValue}'
-      }, {
-        br: '<br />',
-        scheduledEventsValue: marker.scheduledEvents.map(mlEscape).join('<br/>')
+      tooltipData.push({
```

Comment:
The layout for scheduled events need some tweaking I think:

![image](https://user-images.githubusercontent.com/7405507/66819890-9f845380-ef37-11e9-8144-181b8f8d3c67.png)

Reply:
Added a fix in 002ccfb673 to better handle multiple events within a tooltip. There's not much I can do in the case above, but the updated `max-width: 512px` might help.